### PR TITLE
spec(core): ldml: merge keys into key2 🙀

### PR DESCRIPTION
**Pull request by srl295**
- rename keys to key2.kmap (vkey+mod -> key)
- drop key2.keys.vkey

#7042

At some point we may want to rename `key2` to `keys`, but let's just do one thing at a time.

@keymanapp-test-bot skip

### CODE REVIEW

**Review by rc-swag — APPROVED** [review]
Just one comment typo. 
@mcdurdin  may have input.
LGTM

File: core/include/ldml/keyboardprocessor_ldml.ts [hunk 1]
```diff
@@ -225,28 +224,9 @@ class Constants {
   readonly finl_flags_error = 0x0001;
 
   /* ------------------------------------------------------------------
-    * keys section
+    * keys section is now key2.jmap
```
Comment (mcdurdin) [inline core/include/ldml/keyboardprocessor_ldml.ts:227]:
```suggestion
    * keys section is now key2.kmap
```

**Review by mcdurdin — APPROVED** [review]
LGTM.

Thank you @rc-swag for suggesting this. `keys` vs `key2` was a red flag as soon as we introduced it. I commend the idea of renaming `key2` to `keys` soon, so that this little side-track disappears altogether prior to release.

**Comment by srl295** [comment]
> LGTM.
> 
> 
> 
> Thank you @rc-swag for suggesting this. `keys` vs `key2` was a red flag as soon as we introduced it. I commend the idea of renaming `key2` to `keys` soon, so that this little side-track disappears altogether prior to release.

I can see that now. Didn't quite see how it fit together previously - it was the test case that glued it together for me. 

Will rename. Wanted renaming not to confuse things further.

**Pull request merged**